sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/types/PhysicalDataType.scala

```diff
@@ -327,6 +328,17 @@ case class PhysicalStructType(fields: Array[StructField]) extends PhysicalDataTy
   }
 }
 
+class PhysicalVariantType extends PhysicalDataType {
+  private[sql] type InternalType = VariantVal
+  @transient private[sql] lazy val tag = typeTag[InternalType]
+
+  override private[sql] def ordering =
+    throw QueryExecutionErrors.orderedOperationUnsupportedByDataTypeError(
+      "PhysicalVariantType")
```

Comment:
Shall we supports sort in future?

Reply:
I think so, but probably not soon. I also added a todo.